lib/filterx/object-string.h

```diff
@@ -95,7 +95,7 @@ void filterx_string_global_deinit(void);
   { \
     FILTERX_OBJECT_STACK_INIT(string), \
     .str = (cstr), \
-    .str_len = ((cstr_len) < 0 ? strlen(cstr) : (cstr_len)), \
+    .str_len = (((gssize) cstr_len) < 0 ? strlen(cstr) : (cstr_len)), \
```

Comment:
Shouldn't we check for an explicit `-1` here?

Hopefully our strings will never reach `~2^31`, because all subsequent value would be interpreted as negative on a 32-bit platform, but a `-1` check would still be more concrete.